pyghthouse/data/canvas.py

```diff
@@ -1,26 +1,90 @@
 from typing import Union, Iterable
-import numpy as np
-
-
+# import numpy as np
 class PyghthouseCanvas:
 
-    VALID_IMAGE_TYPE = Union[np.ndarray, Iterable, int]
+    VALID_IMAGE_TYPE = Union[Iterable, int]
     IMAGE_SHAPE = (14, 28, 3)
 
     def __init__(self, initial_image=None):
-        self.image = np.zeros(self.IMAGE_SHAPE, dtype=np.ubyte)
+        self.image = self.init_image_array()
         if initial_image is None:
-            self.image[:, :, 0] = 255
+            self.image[:][:][0] = 255
         else:
             self.set_image(initial_image)
 
-    def set_image(self, new_image: VALID_IMAGE_TYPE) -> np.array:
+    """
+    Flattens the list. Works for nested lists too.
+    """
+    def flatten_list(self,element):
+        if(type(element)==list):
+            result = []
+            for element in element:
+                result.extend(self.flatten_list(element))
+            return result
+        else:
+            return [element]
+
+
+    """
+    Creates a nested list in the shape of IMAGE_SHAPE. 
+    number_cb: function
+        A function that returns an integer (or to integer castable) element.
+    """
+    def init_image_array(self,number_cb=lambda: 0):
+        list = []
+        for x in range(self.IMAGE_SHAPE[0]):
+            list_y = []
+            for y in range(self.IMAGE_SHAPE[1]):
+                list_rgb = []
+                for rgb in range(self.IMAGE_SHAPE[2]):
+                    list_rgb.append(int(number_cb()))
+                list_y.append(list_rgb)
+            list.append(list_y)
+        return list
+
+    """
+    Parses user input to image data.
+    new_image: int | list
+        If int provided, sets all values to this number. Must be between 0 and 255.
+        If list provided, it must contain exactly as many elements as the image needs values,
+            but it may be nested in any way.
+    """
+    def new_image_to_image_array(self,new_image: VALID_IMAGE_TYPE) -> list:
```

Comment:
Please make sure to specify the type arguments:

```suggestion
    def new_image_to_image_array(self,new_image: VALID_IMAGE_TYPE) -> list[int]:
```

Reply:
This implementation supports nested lists and all datatypes that can be cast to integers. Thats why im not that sure about list[int]. On the other side, this is python and the type hints are ignored by the interpreter anyway.

Reply:
Then I think it should be `list[Any]`. If you use VSCode I recommend trying to enable strict type checking in the settings, which should highlight these kinds of issues. But to be fair, I don't know how much of the remaining codebase is already passing the strict check.